### DISCUSSION TITLE
Is there any hook that is triggered before the router starts?

### DISCUSSION THREAD

**Discussion by mohit61** [apollographql/router#1595] [Implementation]
Hi, 

I want to do an api call before the router starts, is there any hook in the apollo router that can be used?

**Comment by bnjjj** [comment]
Not really, you could use the router as a library and do whatever you want before running the router. Like here for example https://github.com/apollographql/router/blob/main/examples/hello-world/src/main.rs#L17

But I'm not sure it would be a good solution.

**Comment by o0Ignition0o** [comment]
Depending on where the router is being deployed (like Docker for example), an entrypoint or a script would be less involved than building your own router
